framework/Models/ExternalModel.py

```diff
@@ -226,7 +259,10 @@ def _externalRun(self, Input, modelVariables):
       if key in self.modelVariableType.keys() or key in additionalKeys:
         InputDict[key] = Input[key]
 
-    self.sim.run(externalSelf, InputDict)
+    if 'current_time' in Input:
```

Comment:
this seems like a very implicit check; is there not a way to know ahead whether we're running in co-solve or independent solve mode? I assume that's what this means.

Reply:
... is this even part of this PR? This looks like it belongs with time-step cross-model communication.